src/build.cc

```diff
@@ -316,16 +316,16 @@ bool Plan::AddSubTarget(Node* node, vector<Node*>* stack, string* err) {
   return true;
 }
 
-bool Plan::CheckDependencyCycle(Node* node, const vector<Node*>& stack,
+bool Plan::CheckDependencyCycle(const Node* node, const vector<const Node*>& stack,
                                 string* err) {
-  vector<Node*>::const_iterator start = stack.begin();
+  vector<const Node*>::const_iterator start = stack.begin();
```

Comment:
Instead, would `auto start = stack.cbegin();` be nor succinct?

Reply:
IIRC `cbegin()` returns `vector<Node* const>::…` though, not `vector<const Node*>::…`.